pulp_2_tests/tests/rpm/api_v2/test_copy.py

```diff
@@ -459,6 +487,47 @@ def copy_units(self, recursive, recursive_conservative, old_dependency):
         )
         return self.client.get(repos[1]['_href'], params={'details': True})
 
+    def copy_and_filter_units(self):
+        """
+        Copy units using ``recursive_conservative`` filtering OUT walrus-5.21.
+        (see Pulp issue #6151 for the reproducer this is based on)
+        """
+        repos = []
+        body = gen_repo(
+            importer_config={'feed': RPM_UNSIGNED_FEED_URL},
+            distributors=[gen_distributor()],
+        )
+        repos.append(self.client.post(REPOSITORY_PATH, body))
+        self.addCleanup(self.client.delete, repos[0]['_href'])
+        sync_repo(self.cfg, repos[0])
+        repos.append(self.client.post(REPOSITORY_PATH, gen_repo()))
+        self.addCleanup(self.client.delete, repos[1]['_href'])
+
+        self.client.post(
+            urljoin(repos[1]['_href'], 'actions/associate/'),
+            {
+                'source_repo_id': repos[0]['id'],
+                'override_config': {
+                    'recursive_conservative': True,
+                },
+                'criteria': {
+                    'type_ids': ['rpm'],
+                    'filters': {
+                        'unit': {
+                            '$and': [
+                                {'filename': {'$exists': True}},
+                                {'$nor':[
+                                    {'$or':[
+                                        {'filename': 'walrus-5.21-1.noarch.rpm'}
+                                    ]}
+                                ]}
+                            ]
+                        }
+                    }
+                },
+            },
+        )
+        return self.client.get(repos[1]['_href'], params={'details': True})
 
 @pytest.mark.recursive_conservative
```

Comment:
Add this annotation -- it will add it to the subset of depsolving tests that make it easy to say "just run the depsolving tests"

Reply:
The class itself is already so labelled - every test here is about depsolving

Reply:
You're right, I didn't see that!